server/graphql/reports/src/reports.rs

```diff
@@ -80,9 +82,21 @@ impl ReportNode {
     pub async fn name(&self) -> &str {
         &self.row.name
     }
+
     pub async fn context(&self) -> ReportContext {
         ReportContext::from_domain(&self.row.context)
     }
+
+    pub async fn context2(&self) -> &Option<String> {
```

Comment:
`context` by itself is fine as a name, but `context2` is taking it too far! 
what are the context and context2 for? why are there two of them? why is one a `ReportContext` and one a `String`

the answers might come later.. but just reading the PR top to bottom I wonder if the column name could be more meaningful

Reply:
Yes it would be good if we could find a better solution or term. Here is the problem maybe you have another idea?

Currently I need to print a patient report for a certain program (e.g. `context="Patient"` and `context2="ABCProgram"`). The patient report for a certain program should only be applicable for patient that are enrolled in this program. Currently, the report assumes that the patient is enrolled in the program and if not the report generation fails. However, it shouldn't be possible to print the report in the first place when you are not enrolled.

Additionally, you could have other Program reports, e.g. a store wide program report which shouldn't show up in the list of patient reports...

Further question, what if further nesting needed? e.g. context3?

We could hack something like having a context "Patient_ABCProgram" or "Store_ABCProgram" and the UI needs to know how to build these contexts. But not the nicest solution...

Reply:
tricky. So the `context`s are hierarchical?  and therefore `context2` is a sub-context or refinement of the report context? The context2 could be named as a child / refinement / subcontext - or the original `ReportContext` could be an object with children under it?

Or the contexts are a set which all apply? in which case an array might make sense?

Reply:
Good question, I create an issue and will have another think about it over the weekend

https://github.com/openmsupply/open-msupply/issues/1059